src/dune_rules/melange/melange_rules.ml

```diff
@@ -1,17 +1,48 @@
 open Import
 open Memo.O
 
+module Output_kind : sig
+  type t =
+    | Private_library_or_emit of Path.Build.t
+    | Public_library of
+        { lib_dir : Path.t
+        ; output_dir : Path.Build.t
+        }
+
+  val [@ocaml.warning "-32"] to_dyn : t -> Dyn.t
```

Comment:
Doesn't `let[@ocaml.warning "-32"]` also work? That might avoid the signature.

Reply:
cleaned up in https://github.com/ocaml/dune/pull/11748